scripts/biomedical/diseasesAtJensenLab/scripts/format_disease_jensen_lab.py

```diff
@@ -0,0 +1,383 @@
+
+    Raises:
+        ValueError: If the string contains any illegal characters.
+    """
+    for char in LIST_ILLEGAL_CHARACTERS:
+    	if char in s:
+    		raise ValueError(f"Illegal character '{char}' found in string: {s}")
+
+
+def check_for_dcid(row):
+	"""Checks dcid columns for illegal characters."""
+	for cell in [row['dcid'], row['DOID_dcid'], row['ICD10_dcid'], row['Gene_dcid']]:
+		check_for_illegal_char(str(cell))
+	return row
+
+
+def format_disease_gene_cols(df, data_type):
+	"""
+	 Formats disease/gene columns, generating DCIDs and names - 
+	 handling formatting for import into a data commons knowledge graph.
+	 """
+	# handle exceptions in Gene names
+	df['Gene'] = df['Gene'].map(HGNC_DICT).fillna(df['Gene'])
+	df['Gene'] = df['Gene'].str.replace('@', '_Cluster')
+
+    # generate gene dcid
+	df['Gene_dcid'] = 'bio/' + df['Gene'].astype(str)
+
+    # generate mappings to existing Disease or ICD10 nodes
+	df['DOID_dcid'] = 'bio/' + df['DOID'].str.replace(':', '_')
+	df['DOID_dcid'] = df['DOID_dcid'].replace('bio/nan', np.nan)
+	df['ICD10_dcid'] = df['ICD10'].str.replace(':', '/')
+
+    # use Disease nodes as default diseaseID mapping
+    # use ICD10 nodes as diseaseID mapping in cases where Disease mapping does not exist
+	df['Disease_dcid'] = df['DOID_dcid']
+	df['Disease_dcid'] = df['Disease_dcid'].fillna(df['ICD10_dcid'])
+	df['Disease_dcid'] = 'dcid:' + df['Disease_dcid']
+
+    # generate dcid for DiseaseGeneAssociation nodes
+	df['dcid'] = 'bio/' + df['Disease'].astype(str).str.replace(':', '_', regex=False) + '_' + df['Gene'].astype(str) + '_association'
+
+    # format name for DiseaseGeneAssociation nodes
+	df['name'] = df['Disease_Name'].astype(str) + ' and ' + df['Gene'].astype(str) + ' association'
+
+    # drop columns not used in import
+	df = df.drop(['Disease'], axis=1)
+	return df
+
+
+def format_RNA_type(df_non_coding):
+    """
+    Categorizes non-coding RNAs based on their gene names and assigns corresponding RNA types.
+
+    This function adds an 'RNA_type' column to the input DataFrame, classifying 
+    non-coding RNAs into specific categories based on keywords found in the 'Gene' column.  
+    It uses a hierarchical approach, checking for specific RNA types first before 
+    assigning a more general type.
+
+    Args:
+        df_non_coding: The input DataFrame containing non-coding RNA data, 
+                       expected to have a 'Gene' column.
+
+    Returns:
+        The modified DataFrame with the added 'RNA_type' column.
+    """
+
+    gene_list = ['orf', 'ZNF']  # Keywords for protein-coding genes (treated as "Gene")
+    sno_rna_list = ['sno', 'SNOR', 'SCAR']  # Keywords for small nucleolar RNAs
+    linc_rna_list = ['LINC', 'linc', 'MIR']  # Keywords for long intergenic non-coding RNAs
+
+    # Default RNA type (Long Non-Coding RNA)
+    df_non_coding['RNA_type'] = 'dcs:NonCodingRNATypeLongNonCodingRNA'
+
+    # Assign more specific RNA types based on gene name keywords
```

Comment:
nit: it might be cleaner to have a list of the rna types and the keywords for each rna type & just do a loop like

```
RNA_TYPES_INFO = [{'keyword': <keyword>, 'type': <type>}, ...]
for rna_type in RNA_TYPES_INFO:
  df_non_coding['RNA_type'] = np.where(
        df_non_coding["Gene"].str.contains(rna_type['keyword']),
        rna_type['type'],
        df_non_coding['RNA_type']
    )
```